solutions/usaco-944.mdx

```diff
@@ -0,0 +1,103 @@
+---
+id: usaco-944
+source: USACO Silver 2019 US Open
+title: Fence Planning
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_fenceplan_silver_open19.html)
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+  
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class FencePlanning {
+	static Cow[] cows;
+	static List<Integer>[] graph;
+	static boolean[] visited;
+	static int lowX = Integer.MAX_VALUE;
+	static int highX = Integer.MIN_VALUE;
+	static int lowY = Integer.MAX_VALUE;
+	static int highY = Integer.MIN_VALUE;
+
+	static class Cow { // class to keep track of the x and y
+		int x;
+		int y;
+	}
+
+	static void floodfill (int currentCow) { // dfs flood fill
+		visited[currentCow] = true; // mark as visited
+		Cow cow = cows[currentCow];
+
+		lowX = Integer.min(lowX, cow.x); // take minimum and maximums
+		highX = Integer.max(highX, cow.x);
+		lowY = Integer.min(lowY, cow.y);
+		highY = Integer.max(highY, cow.y);
+
+		for (int connectedCow : graph[currentCow]) { // for each connected cow
+			if (! visited[connectedCow]) {
+				floodfill(connectedCow);
+			}
+		}
+	}
+
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("fenceplan");
+
+		int n = io.nextInt();
+		int m = io.nextInt();
+		visited = new boolean[n + 1]; // one -indexed
```

Comment:
why not make it 0-indexed

Reply:
i like to make my graphs one indexed because the connections from the input are always given as one-indexed, meaning easier to debug because easier to follow the input